packages/table/src/react/index.js

```diff
@@ -100,27 +110,44 @@ function ColumnHeader(props) {
   if (props.flex && !style.flex) style.flex = props.flex
 
   const handleClick = props.onClick
-    ? evt => props.onClick(getToggledSort(props), evt)
+    ? evt =>
+        props.onClick(
+          ...[evt, props.sort && getToggledSort(props)].filter(Boolean)
+        )
     : undefined
-
-  return (
-    <Tag
-      role="columnheader"
-      {...styles.columnHeader(themeName, props, { active })}
-      {...filterReactProps(props, { tagName: Tag })}
-      onClick={handleClick}
-      style={style}
-      aria-sort={
+  const children = props.onClick ? (
+    <button {...styles.columnHeaderButton()} onClick={handleClick}>
```

Comment:
Not sure if we shouldn't just leave this on the TableColoumn header div

Reply:
do you mean having the onClick event on the columnheader div. that seems fine to me